### PR TITLE
Updating dependencies to latest versions in NuGet

### PR DESCRIPTION
**Pull request by joperezr**
Since I'm not sure what the state of dependabot is, I manually did a run to check all packages that have new versions in NuGet and the following were the ones found. @eerhardt I'd appreciate a look to ensure that the updates seem reasonable.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4203)

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -150,15 +150,15 @@
     <PackageVersion Include="Testcontainers.Redis" Version="$(TestcontainersPackageVersion)" />
     <PackageVersion Include="Testcontainers.Nats" Version="$(TestcontainersPackageVersion)" />
     <!-- playground apps dependencies -->
-    <PackageVersion Include="Dapr.AspNetCore" Version="1.13.0" />
-    <PackageVersion Include="Microsoft.Orleans.Clustering.AzureStorage" Version="8.1.0-nightly.20240307.1" />
```
Comment (danmoseley) [inline Directory.Packages.props:154]: is this behind "8.1.0" or ahead?

File: Directory.Packages.props [hunk 2]
```diff
@@ -150,15 +150,15 @@
...
+    <PackageVersion Include="Dapr.AspNetCore" Version="1.13.1" />
+    <PackageVersion Include="Microsoft.Orleans.Clustering.AzureStorage" Version="8.1.0" />
+    <PackageVersion Include="Microsoft.Orleans.Persistence.AzureStorage" Version="8.1.0" />
+    <PackageVersion Include="Microsoft.Orleans.Client" Version="8.1.0" />
+    <PackageVersion Include="Microsoft.Orleans.Server" Version="8.1.0" />
+    <PackageVersion Include="Microsoft.Orleans.Sdk" Version="8.1.0" />
     <!-- Pinned version for Component Governance - Remove when root dependencies are updated -->
     <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageVersion Include="Newtonsoft.Json.Schema" Version="3.0.15" />
+    <PackageVersion Include="Newtonsoft.Json.Schema" Version="3.0.16" />
```
Comment (danmoseley) [inline Directory.Packages.props:161]: remove commented line below?

File: Directory.Packages.props (hunk 1 above)
Comment (joperezr) [reply Directory.Packages.props:154]: this is going from 8.1.0 unstable, to 8.1.0 stable, so the update is newer (ahead)

**Review by danmoseley — APPROVED** [review]
Seems fine to me with the caveat how do we know if any of the major changes eg KubernetesClient are too breaking. Update and see if anyone complains?

We should have a markdown in the repo that describes our policy.

**Comment by joperezr** [comment]
I don't have a script, but we actually do want latest of all of those dependencies (since I'm actually getting latest in NuGet for all of these) so I can still take care of this with one single queue of a build (which I'm about to do)

File: NuGet.config [hunk 3]
```diff
@@ -26,6 +26,7 @@
     </packageSource>
     <packageSource key="dotnet-public">
       <package pattern="*" />
+      <package pattern="Microsoft.Orleans.*" />
```
Comment (eerhardt) [inline NuGet.config:29]: Can we remove the orleans-nightly?

Comment (eerhardt) [reply NuGet.config:29]: If we use it again, we can add it again. Might as well remove things we aren't using.

Comment (joperezr) [reply NuGet.config:29]: Agreed. Removed.

File: src/Components/Aspire.Azure.Storage.Blobs/ConfigurationSchema.json [hunk 4]
```diff
@@ -77,6 +77,29 @@
                           "format": "uri",
                           "description": "Gets or sets the secondary storage 'System.Uri' that can be read from for the storage account if the\naccount is enabled for RA-GRS.\n\nIf this property is set, the secondary Uri will be used for GET or HEAD requests during retries.\nIf the status of the response from the secondary Uri is a 404, then subsequent retries for\nthe request will not use the secondary Uri again, as this indicates that the resource\nmay not have propagated there yet. Otherwise, subsequent retries will alternate back and forth\nbetween primary and secondary Uri."
                         },
+                        "Request100ContinueOptions": {
```
Comment (joperezr) [inline src/Components/Aspire.Azure.Storage.Blobs/ConfigurationSchema.json:80]: @eerhardt  FYI, looks like the new version of Azure storage now has a new config setting.

Comment (joperezr) [reply src/Components/Aspire.Azure.Storage.Blobs/ConfigurationSchema.json:80]: cc @sebastienros

Comment (eerhardt) [reply src/Components/Aspire.Azure.Storage.Blobs/ConfigurationSchema.json:80]: Yep - looks like it came in with https://github.com/Azure/azure-sdk-for-net/pull/43999.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**